testing/tests/DevExpress.ui.widgets/overlay.tests.js

```diff
@@ -780,31 +780,156 @@ testModule('visibility', moduleConfig, () => {
-        const onShown = sinon.stub();
-        const $overlay = $('#overlay').dxOverlay({
-            onShowing: e => e.cancel = true,
-            onShown
+    testModule('e.cancel', {
+        beforeEach: function(assert) {
+            this.onShown = sinon.stub();
+            this.onHidden = sinon.stub();
+            this.$overlay = $('#overlay').dxOverlay({
+                onShown: this.onShown,
+                onHidden: this.onHidden,
+            });
+            this.overlay = this.$overlay.dxOverlay('instance');
+            this.checkIsNotShown = () => {
+                assert.notOk(this.onShown.called, 'onShown should not be called');
+            };
+            this.checkIsShown = () => {
+                assert.notOk(this.$overlay.is(':hidden')), 'overlay is not hidden';
+                assert.ok(this.onShown.called, 'onShown should be called');
+            };
+            this.checkIsNotHidden = () => {
+                assert.notOk(this.$overlay.is(':hidden')), 'overlay is not hidden';
+                assert.notOk(this.onHidden.called, 'onHidden should not be called');
+            };
+            this.checkIsHidden = () => {
+                assert.ok(this.$overlay.is(':hidden')), 'overlay is hidden';
+                assert.ok(this.onHidden.called, 'onHidden should be called');
+            };
+        }
+    }, () => {
+        test('overlay should not be shown if e.cancel == true in the onShowing event handler (T825865)', function(assert) {
+            this.overlay.option({ onShowing: e => e.cancel = true });
+            this.overlay.show();
+
+            assert.ok(this.$overlay.is(':hidden')), 'overlay is hidden';
+            this.checkIsNotShown();
         });
-        const overlay = $overlay.dxOverlay('instance');
-        overlay.show();
 
-        assert.ok($overlay.is(':hidden')), 'overlay is hidden';
-        assert.notOk(onShown.called, 'onShown should not be called');
-    });
+        test('overlay should not be hidden if e.cancel == true in the onHidinging event handler', function(assert) {
+            this.overlay.option({
+                visible: true,
+                onHiding: e => e.cancel = true
+            });
+            this.overlay.hide();
 
-    test('overlay should not be hidden if e.cancel == true in the onHidinging event handler', function(assert) {
-        const onHidden = sinon.stub();
-        const $overlay = $('#overlay').dxOverlay({
-            visible: true,
-            onHiding: e => e.cancel = true,
-            onHidden
+            this.checkIsNotHidden();
+        });
+
+        test('overlay should be shown after timeout if e.cancel == promise resolving false after timeout', function(assert) {
+            const done = assert.async();
+            this.overlay.option({
+                onShowing: e => e.cancel = new Promise((resolve) => { setTimeout(() => resolve(false), 0); }),
```

Comment:
please refactor all such places. It's hard to read this when it's written in a single line

Reply:
done